Reduce default MBEDTLS_MPI_MAX_SIZE

RonEld opened the pull request:
### Description

<!-- 
    Required
    Add here detailed changes summary, testing results, dependencies 
    Good example: https://os.mbed.com/docs/latest/reference/workflow.html (Pull request template)
-->
Reduce the default size of `MBEDTLS_MPI_MAX_SIZE` to 512 bytes,
as the default 1024 consumes much stack, and supporting RSA 4096 bit
may suffice at the moment.


### Pull request type

<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

Commit:
Reduce default MBEDTLS_MPI_MAX_SIZE

Reduce the default size of `MBEDTLS_MPI_MAX_SIZE` to 512 bytes,
as the default 1024 consumes much stack, and supporting RSA 4096 bit
may suffice at the moment.

0xc0170 commented:
Travis failure will be fixed on master soon (we will request rebase once done). See #8945

Patater commented on features/mbedtls/importer/adjust-config.sh:
```diff
@@ -140,6 +140,8 @@ conf unset MBEDTLS_SSL_TRUNCATED_HMAC
 
 conf unset MBEDTLS_PLATFORM_TIME_TYPE_MACRO
 
+conf set MBEDTLS_MPI_MAX_SIZE     512
```
May be worth adding a comment on effects of this limitation, as well as its benefits.

RonEld replied:
I added a comment, as asked

Patater reviewed (approved):
LGTM. Could use a comment.

Commit:
Add comment for reasoning for changing default

Add a comment in the `adjust-config.sh` script, for effects
and benefits of the new value.

0xc0170 commented:
> Travis failure will be fixed on master soon (we will request rebase once done). See #8945

The fix is on master, if you rebase, it should allow travis to run

RonEld commented:
OK, I'll rebase. Do you want me to squash the two commits while I'm on it?

cmonr commented:
@RonEld No need to squash. Both commits look good asis.

Patater reviewed (approved):
Note that our TLS Client example uses 256 here. https://github.com/ARMmbed/mbed-os-example-tls/blob/master/tls-client/mbedtls_entropy_config.h#L36

cmonr commented:
CI started

RonEld commented:
@Patater 
>Note that our TLS Client example uses 256 here.

Thanks for the reminder! I have  created https://github.com/ARMmbed/mbed-os-example-tls/pull/220 to avoid future compilation warnings.

RonEld commented:
@cmonr @0xc0170  Why did the CI fail? I can't seem to find in the log a failure

0xc0170 commented:
Will restart CI once rc 2 is completed (the failure is new, not related, we will track it and report back)

0xc0170 commented:
CI restarted
